report_thesis/src/sections/proposed_approach/testing_validation.tex

```diff
@@ -47,76 +59,142 @@ \subsection{Validation and Testing Procedures for Model Evaluation}\label{subsec
+Essentially, this step enables the partitioning steps that follow, by ensuring each data item has an associated fold number.
+Following this, the dataset is divided into a test set, which always consists of the data points assigned to fold $k-1$, and the remaining data forms the training set, as outlined in Line~\ref{line:split_dataset}.
+The training data is further divided into $k-1$ sets for cross-validation. 
+For each fold $i$ where $i \in \{0, 1, \ldots, k-2\}$, we create a cross-validation training set $\mathbf{T}_\text{train}[i]$ by excluding the $i$-th fold from the set of $k-1$ folds, and use the $i$-th fold as the validation set $\mathbf{T}_\text{val}[i]$.
+These pairs of training and validation sets are then appended to the list of cross-validation sets $\mathbf{T}_\text{cv}$ (Line~\ref{line:create_folds}).
 
 Finally, the fold indicator column is removed from all datasets before returning the final partitions (Line~\ref{line:remove_fold_column}).
 The fold indicator column was added to keep track of which data points belong to which folds, which is crucial for ensuring that data points are correctly partitioned into their respective training and test sets during cross-validation. 
 This cleanup step ensures that the fold information does not interfere with subsequent data processing or model training.
 
 The final output of this procedure consists of:
 \begin{itemize}
-    \item The cross-validation folds \(\mathbf{F}_\text{cv}\), each containing a tuple of training and validation sets.
-    \item The training set \(\mathbf{D}_\text{train}\) in its entirety.
+    \item A set of tuples \(\mathbf{T}_\text{cv}\), where each tuple contains a training set and a validation set.
+    \item The overall training set \(\mathbf{D}_\text{train}\), consisting of all the data points not in the test set.
     \item The test set \(\mathbf{D}_\text{test}\), distinct from the training set.
 \end{itemize}
 
-The data partitioning does not modify the original dataset, it merely partitions it.
-For that reason, each of the datasets that are returned have the same structure as shown in Table~\ref{tab:final_dataset_example}.
+The data partitioning does not modify the original dataset; it merely partitions it.
+For that reason, each of the datasets that are returned has the same structure as shown in Table~\ref{tab:final_dataset_example}.
+
+Given that the data partitioning procedure aims to distribute extreme concentration values evenly among the training folds while minimizing their presence in the test set, it is crucial to determine an optimal value of $p$ that minimizes the number of extreme values in the test set while still maintaining its general representativeness.
+By general representativeness, we mean ensuring that the test set reflects the general distribution of the dataset without being skewed by extreme values.
+This balance is essential for accurately assessing the model's performance on typical data points.
```

Comment:
I dont know how, but I feel like this could shortened down a fair bit. Its a lot of text to say that $p$ has a big influence on minimizing the extreme values in the test set.

Reply:
I'm also unsure how.

It's not just that it's an important variable to optimize to minimize the extreme values. That would be trivial. Just set it really high.

The point is that you have to _maximize_ the _general_ representativeness while _minimizing_ the amount of extreme values present in the test set.

Also don't want to spend a ton more time on this sentence, as defining this + the places that rely on it took a while... as the trade-offs section relies on `general representativeness` as well.